src/System.CommandLine/Binding/ArgumentConverter.cs

```diff
@@ -323,5 +350,45 @@ internal static T GetValueOrDefault<T>(this ArgumentConversionResult result)
                     return default!;
             }
         }
+
+        public static bool TryConvertBoolArgument(ArgumentResult argumentResult, out object? value)
```

Comment:
Is there a reason the out parameter is `object?` not `bool`?

```suggestion

        public static bool TryConvertBoolArgument(ArgumentResult argumentResult, out bool value)
```

Reply:
Good catch! Just a refactoring artifact.